sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -1109,21 +1145,33 @@ case class ShowCreateTableCommand(table: TableIdentifier)
           )
         }
 
-        if (tableMetadata.tableType == VIEW) {
-          throw new AnalysisException("Hive view isn't supported by SHOW CREATE TABLE")
-        }
-
         if ("true".equalsIgnoreCase(tableMetadata.properties.getOrElse("transactional", "false"))) {
           throw new AnalysisException(
             "SHOW CREATE TABLE doesn't support transactional Hive table. " +
               s"Please use `SHOW CREATE TABLE ${tableMetadata.identifier} AS SERDE` " +
               "to show Hive DDL instead.")
         }
 
-        convertTableMetadata(tableMetadata)
+        if (tableMetadata.tableType == VIEW) {
```

Comment:
I think we should put it in a higher level
```
      val metadata = if (DDLUtils.isDatasourceTable(tableMetadata) || tableMetadata.tableType == VIEW) {
        tableMetadata
      } else {
        ...
```

Reply:
We need to check for unsupported feature like `partitioned view` in this branch.

Reply:
ah makes sense